anker/src/processor.rs

```diff
@@ -66,35 +74,73 @@ fn process_initialize(program_id: &Pubkey, accounts_raw: &[AccountInfo]) -> Prog
     )?;
 
     // Create and initialize an stSOL SPL token account for the reserve.
-    let reserve_account_seeds = [
+    let st_sol_reserve_account_seeds = [
         anker_address.as_ref(),
-        ANKER_RESERVE_ACCOUNT,
-        &[reserve_account_bump_seed],
+        ANKER_STSOL_RESERVE_ACCOUNT,
+        &[st_sol_reserve_account_bump_seed],
     ];
     create_account(
         &spl_token::ID,
         &accounts,
-        accounts.reserve_account,
+        accounts.st_sol_reserve_account,
+        &rent,
+        spl_token::state::Account::LEN,
+        &st_sol_reserve_account_seeds,
+    )?;
+    initialize_spl_account(
+        &accounts,
+        &st_sol_reserve_account_seeds,
+        accounts.st_sol_reserve_account,
+        accounts.st_sol_mint,
+    )?;
+
+    // Create and initialize an UST SPL token account for the reserve
+    let ust_reserve_account_seeds = [
+        anker_address.as_ref(),
+        ANKER_UST_RESERVE_ACCOUNT,
+        &[ust_reserve_account_bump_seed],
+    ];
+    create_account(
+        &spl_token::ID,
+        &accounts,
+        accounts.ust_reserve_account,
         &rent,
         spl_token::state::Account::LEN,
-        &reserve_account_seeds,
+        &ust_reserve_account_seeds,
+    )?;
+    initialize_spl_account(
+        &accounts,
+        &ust_reserve_account_seeds,
+        accounts.ust_reserve_account,
+        accounts.ust_mint,
     )?;
-    initialize_reserve_account(&accounts, &reserve_account_seeds)?;
+
+    let (_, token_swap_bump_seed) =
+        Pubkey::find_program_address(&[&accounts.pool.key.to_bytes()], &spl_token_swap::id());
 
     let anker = Anker {
```

Comment:
Note to self: I wanted to ask, don’t we need to store the UST mint address somewhere? But no we don’t, we can always extract it from our UST reserve. And we trust the UST reserve, because it lives at a program-derived address that we verify.